docs/tutorial_kp.md

```diff
@@ -0,0 +1,283 @@
+       --registry https://index.docker.io/v1/ \
+       --registry-user my-dockerhub-username \
+       -n default
+    ```
+
+   > Note: Learn more about kpack secrets with the [kpack secret documentation](secrets.md)
+
+1. Create a cluster store configuration
+
+   A store resource is a repository of [buildpacks](http://buildpacks.io/) packaged
+   in [buildpackages](https://buildpacks.io/docs/buildpack-author-guide/package-a-buildpack/) that can be used by kpack
+   to build images. Later in this tutorial you will reference this store in a Builder configuration.
+
+   We recommend starting with buildpacks from the [paketo project](https://github.com/paketo-buildpacks). The example
+   below pulls in java buildpack from the paketo project.
+
+    ```
+    kp clusterstore save default -b gcr.io/paketo-buildpacks/java
+    ```
+
+   > Note: Buildpacks are packaged and distributed as buildpackages which are docker images available on a docker registry. Buildpackages for other languages are available from [paketo](https://github.com/paketo-buildpacks).
+
+1. Create a cluster stack configuration
+
+   A stack resource is the specification for
+   a [cloud native buildpacks stack](https://buildpacks.io/docs/concepts/components/stack/) used during build and in the
+   resulting app image.
+
+   We recommend starting with the [paketo base stack](https://github.com/paketo-buildpacks/stacks) as shown below:
+
+    ```
+    kp clusterstack save base --build-image paketobuildpacks/build:base-cnb --run-image paketobuildpacks/run:base-cnb
+    ```
+
+1. Create a Builder configuration
+
+   A Builder is the kpack configuration for a [builder image](https://buildpacks.io/docs/concepts/components/builder/)
+   that includes the stack and buildpacks needed to build an image from your app source code.
+
+   The Builder configuration will write to the registry with the secret configured in step one and will reference the
+   stack and store created in step three and four. The builder order will determine the order in which buildpacks are
+   used in the builder.
+
+    ```
+    kp builder save my-builder \
+      --tag <DOCKER-IMAGE-TAG> \
+      --stack base \
+      --store default \
+      --buildpack paketo-buildpacks/java \
+      -n default
+    ```
+
+    - Make sure to replace `<DOCKER-IMAGE>` with the tag in the registry you configured in step #1. Something like:
+      your-name/builder or gcr.io/your-project/builder
+
+1. Apply a kpack image configuration
+
+   An image configuration is the specification for an image that kpack should build and manage.
+
+   We will create a sample image that builds with the builder created in step five.
+
+   The example included here utilizes
+   the [Spring Pet Clinic sample app](https://github.com/spring-projects/spring-petclinic). We encourage you to
+   substitute it with your own application.
+
+   Create an image configuration:
+
+    ```yaml
+    kp image save tutorial-image \
```

Comment:
--wait?

Reply:
I liked how the next steps shows you how to check the status while building and also how to get the logs with `kp build logs`